File: python/leap_years.py
### Leap Years in Python
#### https://code.golf/leap-years#python
## Copyright(c) Kira K. 2023
### To be optimised for Code Golf

### Unlike most of these, this was just adapted from elsewhere
for y in range(1800,2401):
    if(y%4<1 and y%100>0 or y%400<1):
        print(y)

File: java/leapyears.java
/*** Leap Years in Java
**** https://code.golf/leap-years#java
* Copyright (c) Kira K. 2023
** To be optimised for Code Golf */
// Pre-Golf:
class Main {
    public static void main(String[]args){
        for (int year = 1800; year <= 2400; year++){
            if (year % 4 == 0 && (year % 100 != 0 || year % 400 == 0)){
                System.out.println(year);
            }
        }
    }
    int x100(int n){
        return n*100;
    }
}

// Post-Golf:
interface A{static void main(String[]s){for(int y=1800;y<=2400;y++){if(y%4<1&y%100>0|y%400<1){System.out.println(y);}}}}